imaging/ml/ml_codelab/breast_density_cloud_ml.ipynb

```diff
@@ -1,911 +1,930 @@
+    "We need to create a Cloud Healthcare API Dataset and Dicom Store to store the the DICOM instances sourced from TCIA. Please refer [here](https://cloud.google.com/healthcare/docs/introduction) for a description of the Cloud Healthcare API data hierarchy. Add your parameters for Cloud Healthcare API below:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "NSr5StGBZkYd"
+   },
+   "outputs": [],
+   "source": [
+    "project_id = \"MY_PROJECT\" # @param\n",
+    "location = \"us-central1\" # @param\n",
+    "dataset_id = \"MY_DATASET\" # @param\n",
+    "dicom_store_id = \"MY_DICOM_STORE\" # @param\n",
+    "\n",
+    "# Input data used by AutoML must be in a bucket with the following format.\n",
+    "automl_bucket_name= \"gs://\" + project_id + \"-vcm\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "vgBA16ptbacM"
+   },
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import httplib2\n",
+    "import os\n",
+    "from oauth2client.client import GoogleCredentials\n",
+    "\n",
+    "# Path to Cloud Healthcare API.\n",
+    "HEALTHCARE_API_URL = 'https://healthcare.googleapis.com/v1beta1'\n",
+    "\n",
+    "http = httplib2.Http()\n",
+    "http = GoogleCredentials.get_application_default().authorize(http)\n",
+    "\n",
+    "# Create Cloud Healthcare API dataset.\n",
+    "path = os.path.join(HEALTHCARE_API_URL, 'projects', project_id, 'locations', location, 'datasets?dataset_id=' + dataset_id)\n",
+    "headers = {'Content-Type': 'application/json'}\n",
+    "resp, content = http.request(path, method='POST', headers=headers)\n",
+    "assert resp.status == 200, 'error creating Dataset, code: {0}, response: {1}'.format(resp.status, content)\n",
+    "print('Full response:\\n{0}'.format(content))\n",
+    "\n",
+    "# Create Cloud Healthcare API DICOM store.\n",
+    "path = os.path.join(HEALTHCARE_API_URL, 'projects', project_id, 'locations', location, 'datasets', dataset_id, 'dicomStores?dicom_store_id=' + dicom_store_id)\n",
+    "resp, content = http.request(path, method='POST', headers=headers)\n",
+    "assert resp.status == 200, 'error creating DICOM store, code: {0}, response: {1}'.format(resp.status, content)\n",
+    "print('Full response:\\n{0}'.format(content))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "HKsCfbXorosM"
+   },
+   "source": [
+    "Next, we are going to transfer the DICOM instances to the Cloud Healthcare API.\n",
+    "\n",
+    "Note: We are transfering >100GB of data so this will take some time to complete"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "shhPzNFArpHH"
+   },
+   "outputs": [],
+   "source": [
+    "http = httplib2.Http()\n",
```

Comment:
What do you think of putting the reusable code to a script instead?

Reply:
i've used same approach as it was in this files, may be it was made to make communication with api more visual, i may refactor ipynb to be simple calls of methods moved to py files but it would be not as clear for user
in this example we just create request and perform it, we'll have to hide some info about this request, is it ok?